### ISSUE TITLE
Opening links in existing window from the command line

### ISSUE THREAD

**Issue by davewongillies** [skyjake/lagrange#37]
If I open a gemini or gopher link from the command line (or say from another program that calls Lagrange), a whole new session/window of lagrange opens. It'd be great if the link could automatically open as a tab in an existing lagrange window.

**Comment by skyjake** [comment]
Hmm, I'll give this some thought...

Multiple concurrent instances are _not_ handled gracefully at the moment — they will merrily overwrite each others' runtime files (bookmarks, visited URLs, etc.). Implementing this makes it easier to fix that problem as well.

**Comment by skyjake** [comment]
b97462993b00ecc5cd04051b4fc2abd164fefb04 implements an IPC mechanism that enables sending commands to a running instance of Lagrange.

I'll still need to document the options in `--help`.

**Comment by davewongillies** [comment]
Excellent, thank you so much 🥳

**Comment by khlsvr** [comment]
I use a rofi dmenu to launch things and I have noticed the most use I get from just running my script that runs several different executables with ```executable "clipboard content"``` , but with lagrange this doesn't work. The lagrange will not be launching at all if the clipboard content is not a valid URL. For example most web browsers launch a new tab with the search of the "clipboard content" on the set default search engine. I wish lagrange did the same.

behaviour:
```
$  lagrange anything
Invalid URL/file: anything
```
wanted behaviour in my case:
Lagrange launches a new tab in an existing lagrange instance if it exists, otherwise in a new instance, with the page 
gemini://geminispace.info/search?anything

I will throw in here another unrelated small "enhancement" idea or an opinion. With certificates when adding a new identity, I wish the "valid until" had some default value (that you could set maybe in settings even). It's a bit tedious to type in the YYYY-MM-DD HH:MM:SS but I guess we're not creating new identities so often so this is not a problem.

**Comment by skyjake** [comment]
@khlsvr Would it work for you to have a command line option that causes Lagrange to treat the next argument as URL field input, so it can work both with URLs and search queries? I'm not familiar with Rofi but it's always possible to make an alias/script that runs `lagrange` with such an option.

I'm not willing to make this default behavior since that would mean any mistake in the command line arguments automatically means making a search query.

> With certificates when adding a new identity, I wish the "valid until" had some default value (that you could set maybe in settings even). It's a bit tedious to type in the YYYY-MM-DD HH:MM:SS

It is sufficient to write just the year, or YYYY-MM-DD. I should display a help tip about this somewhere in the UI.

**Comment by khlsvr** [comment]
```
#!/bin/bash

ARG=$@

if [ -n "${ARG}" ]
then
    if [[ "$ARG" != "$date" && "$ARG" != kill ]]
    then
        coproc ( $ARG "$(xclip -o)" > /dev/null  2>&1 )
        exec 1>&-
    elif  [ "${ARG}" == "kill" ]
    then 
         coproc ( $(i3-msg kill) > /dev/null  2>&1 )
        #exec i3-msg exec sleep 1 && i3-msg kill
    fi
else
    #echo "$(date)"
    echo "qutebrowser"
    echo "mpv"
    echo "mpv --ytdl-format=best"
    echo "translation_lojban_run"
    echo "text-to-speech"
    echo "translation_speak"
    echo "translation_google"
    echo "translation_japanese"
    echo "feh"
    echo "lagrange"
    echo "kill"
fi
```
That script is run in rofi, rofi gives me a popup of options from that "else" and I then choose one option and it executes it with clipboard content being the argument. So every other time I might want to copy gemini URL to clipboard and other times just a search word, or just run the lagrange if it's not running to access tabs from last session (and just ditch the new tab it opens from some random clipboard content I had.)

So do you mean that if lagrange had a flag like ```lagrange --url <url>``` it would treat it like it does now without flags, and in that version of lagrange if no flags given, it would treat it with search? I guess in that scenario I would need to add another lagrange row to the "else" with a flag.  As it stands now, I could add another "elif" for lagrange to process the argument, but I was just wondering that since it's a common feature in web browsers to evaluate the argument whether it's an URL or a search string that maybe it could be possible to do natively with lagrange too. 

But sure if it's a lot of work to have it process it, then I guess better work on some other issues that may be more important. As for whether the flag helps me I'm not sure how much. If it's useful for some other people too then I guess it's worth going for it.

As for YYYY-MM-DD ok, that's an improvement. However not having to focus that field at all is more than twice the better but yeah, not a big problem :)

**Comment by skyjake** [comment]
Thanks for showing the script, that clarifies things for me.

I meant that there could be a flag like `--url-or-search <text>` that opens a new tab and inserts `<text>` in the URL input field. The behavior would be equivalent to typing `<text>` in the URL field and pressing Enter. So if it's an URL, it gets opened, and if it's not, it goes to the configured search URL as a query parameter.

**Comment by khlsvr** [comment]
Oh ok! In that case it would be very helpful and solve my problem and get instant use.

**Comment by skyjake** [comment]
Added #232 so this is not forgotten.

**Comment by skyjake** [comment]
`--url-or-search` (#232) implemented for v1.5.
